cpp/cert/test/rules/CTR55-CPP/test.cpp

```diff
@@ -20,10 +20,26 @@ void f1(std::vector<int> &v) {
   }
   for (auto i = v.begin(),
             l = (i + std::min(static_cast<std::vector<int>::size_type>(10),
-                              v.size()));
-       i != l; ++i) { // COMPLIANT
+                              v.size())); // NON_COMPLIANT - technically in the
+                                          // calculation
+       i != l; ++i) {                     // COMPLIANT
   }
 
   for (auto i = v.begin();; ++i) { // NON_COMPLIANT
   }
+}
+
+void test_fp_reported_in_374(std::vector<int> &v) {
+  {
+    auto end = v.end();
+    for (auto i = v.begin(); i != end; ++i) { // COMPLIANT
+    }
+  }
+
+  {
+    auto end2 = v.end();
+    end2++;                                    // NON_COMPLIANT
+    for (auto i = v.begin(); i != end2; ++i) { // NON_COMPLIANT[FALSE_NEGATIVE]
```

Comment:
Isn't the access `++i` not guarded by a upperbound check?
Or is there still local data flow `end2` to `end2` in `i !=end2` even though it is modified using `end2++`?

If that is the case, couldn't we invalidate `end2++` by looking at modifications?

Reply:
"Or is there still local data flow end2 to end2 in i !=end2 even though it is modified using end2++?" - yes, that is correct, it would need explicit invalidation modelling

I have made a note in the testcase about not handling this, think we can leave it as a future improvement